Source/Applications/ApplicationArtifactIdentifierStringConverter.cs

```diff
@@ -129,6 +136,9 @@ void ValidateIdentifierString(string identifierAsString)
             var applicationArtifactTypeSeparatorIndex = identifierAsString.IndexOf(ApplicationArtifactTypeSeparator);
```

Comment:
Suggestion for improvement; responsibility wise.. 
Lets move the `.indexOf()` thing into the `ThrowIf...` methods - as it is only being used by the validators themselves and should therefor be isolated in those methods.

Reply:
👍